support/util/imagefetch_test.go

```diff
@@ -0,0 +1,131 @@
+package util
+
+import (
+	"context"
+	"github.com/onsi/gomega"
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	"github.com/openshift/hypershift/support/releaseinfo"
+	fakereleaseprovider "github.com/openshift/hypershift/support/releaseinfo/fake"
+	corev1 "k8s.io/api/core/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/fake"
```

Comment:
Can we please stick to the controller-runtime client to not pick up 20k LOC of dependencies and be consistent?

Reply:
It's just testing dependencies on the fake client and some of the systems use client-go directly so I am hesitant to do that

Reply:
I think it's beneficial to have that anyways as long as we are using client-go in the project which is used at a large scale.

Reply:
I agree with @alvaroaleman, @relyt0925 I know we use the kube client directly in more places but we are heavily controller-runtime based and to continue towards consolidating the code base idiomatically across controllers in that direction.

Could we please just use mgr.GetAPIReader() to Lookup the active containerImage and also drop kube fake client dep?

Reply:
I don't agree with that being in this initial refactor

Reply:
I am fine with that being done in a future refactor that I can also drive but I don't want to keep introducing more changes on this one @enxebre .

Reply:
actually scratch that that isn't bad I can do that

Reply:
back to my original point: yea so I would like the refactor to be performed separate and it doesn't make sense in this initial iteration in my eyes.

I'm also happy to drive that one but I don't believe it should be done in this PR